src/arch/linux.rs

```diff
@@ -16,3 +17,11 @@ pub fn package_installed(package: &str) -> Result<bool> {
 		None
 	))
 }
+
+/// Check if file permissions are secured
+pub fn file_permissions(file: &str, perms: u32) -> Result<bool> {
+	let f = File::open(&file)?;
+	let metadata = f.metadata()?;
+	let current_perms = metadata.permissions().mode();
+	Ok(current_perms == perms + 100000)
```

Comment:
idek whats going on here but it should be `(current_perms & perms) == u32::MAX` or `~(current_perms & perms) == 0`